c_src/libSigPyProcTim.cpp

```diff
@@ -0,0 +1,159 @@
+#include <cmath>
+#include <omp.h>
+#include <fftw3.h>
+#include <complex.h>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#include <libUtils.hpp>
+
+namespace py = pybind11;
+
+
+void runningMedian(py::array_t<float> inarray, py::array_t<float> outarray,
+    int window, int nsamps) {
+    py::buffer_info inbuf = inarray.request(), outbuf = outarray.request();
+
+    float* indata  = (float*)inbuf.ptr;
+    float* outdata = (float*)outbuf.ptr;
+
+    Mediator* m = MediatorNew(window);
+    for (int ii = 0; ii < nsamps; ii++) {
+        MediatorInsert(m, indata[ii]);
+        outdata[ii] = indata[ii] - (float)MediatorMedian(m);
+    }
+}
+
+
+void runningMean(py::array_t<float> inarray, py::array_t<float> outarray, 
+    int window, int nsamps){
+    py::buffer_info inbuf = inarray.request(), outbuf = outarray.request();
+
+    float* indata  = (float*)inbuf.ptr;
+    float* outdata = (float*)outbuf.ptr;
+
+    double sum = 0;
+    for (int ii = 0; ii < nsamps; ii++) {
+        sum += indata[ii];
+        if (ii < window){
+            outdata[ii] = indata[ii] - (float)sum / (ii + 1);
+        }
+        else {
+            outdata[ii] = indata[ii] - (float)sum / (window + 1);
+            sum -= indata[ii - window];
+        }
+    }
+}
+
+void runBoxcar(py::array_t<float> inarray, py::array_t<float> outarray,
+    int window, int nsamps) {
+    py::buffer_info inbuf = inarray.request(), outbuf = outarray.request();
+
+    float* indata  = (float*)inbuf.ptr;
+    float* outdata = (float*)outbuf.ptr;
+
+    double sum = 0;
+    for (int ii = 0; ii < window; ii++) {
+        sum += indata[ii];
+        outdata[ii] = sum / (ii + 1);
+    }
+
+    for (int ii = window / 2; ii < nsamps - window / 2; ii++) {
+        sum += indata[ii + window / 2];
+        sum -= indata[ii - window / 2];
+        outdata[ii] = sum / window;
+    }
+
+    for (int ii = nsamps - window; ii < nsamps; ii++) {
+        outdata[ii] = sum / (nsamps - ii);
+        sum -= indata[ii];
+    }
+}
+
+void downsampleTim(py::array_t<float> inarray, py::array_t<float> outarray, 
+    int factor, int newLen) {
+    py::buffer_info inbuf = inarray.request(), outbuf = outarray.request();
+
+    float* indata  = (float*)inbuf.ptr;
+    float* outdata = (float*)outbuf.ptr;
+
+#pragma omp parallel for default(shared)
```

Comment:
Just checking this works via pybind? (if so cool!)

Reply:
should work. will check #6